### ISSUE TITLE
U-net model have non smooth loss

### ISSUE THREAD

**Issue by mrgloom** [WeidiXie/cell_counting_v2#1]
![3aae123f4336473b801f2cb0d66f8afb](https://cloud.githubusercontent.com/assets/4003908/26413968/5874bd4a-40b6-11e7-8f20-75bd92e66012.png)

U-net model have non smooth loss, can you comment this behaviour?

**Comment by WeidiXie** [comment]
Hi, 

This is because the learning rate schedule in the code,
it is essentially a gradient restart mechanism, to make the network converge to better minima, 
as you can see from this curve, the validation loss is decreased every time you restart the learning rate.

If you don't want that, just change the step_decay function.

Weidi

**Comment by mrgloom** [comment]
![val_loss_graph_buildmodel_u_net](https://cloud.githubusercontent.com/assets/4003908/26424618/9118e90e-40da-11e7-9927-228c9ac33ccd.png)

Yes it helped. In both settings I get about the same result ~3 difference per image.

**Comment by WeidiXie** [comment]
Hi,

Good to hear that. 
The U-net version is a pretty simple network without any fine-tuning 
(mostly because the dateset is too simply to play).
 
But, feel free to edit the code for your application.
